### ISSUE TITLE
Conditionally showing a data table leads to infinite loading

### ISSUE THREAD

**Issue by bloemy7** [FlutterFlow/flutterflow-issues#3211]
### Can we access your project?

- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

I have two data tables. Both are conditionally visible based on the same boolean value, but of opposite condition, so there is only ever one that shows.
If I remove the conditional visibility, both tables show up and work as expected, so clearly they are set up correctly.
With the conditional visibility, the table that shows first does show up correctly, but after clicking the button that hides that table and shows the second one, the second table shows up but infinitely loads - i.e the data inside it never loads.

### Expected Behavior

Expected behavior is for both tables to show up correctly when needed.

### Steps to Reproduce

1. Create a data table, set it up with a Firebase query
2. Duplicate the table, and make the duplicate table "Selectable" 
3. Make the first table only show when boolean X is true
4. Make the second table only show when boolean X is false
5. Add a button which flips the value of boolean X
6. Load the page, see that the first table shows, and then click on the button to flip the state

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

ITEelcmB05NItb0E+YfyL8dvqwMwGkQ5a7sviu5+FBQaI+P2P+x2P+POdhRpSOqHTVVUOVSYuFEG+t7uvYaSEMIoKSerbqZh/80IUxDhTkGjRciVBqq7dHx7N/JTJ3XDyqbYggolObZvWk0twV2EHeiAb3qCf9qOYwx5e6fDbOY=

### Visual documentation

<img width="980" alt="Screenshot 2024-06-24 at 6 43 01 PM" src="https://github.com/FlutterFlow/flutterflow-issues/assets/1565896/1d37000b-ae0d-4ad1-8084-72e356e75576">

### Environment

```markdown
- FlutterFlow version: v.4.1.61+
- Platform: Web
- Browser name and version: Chrome 125
- Operating system and version affected: All
```

### Additional Information

This is blocking us from releasing an update

**Comment by rzambroni** [comment]
Hey @bloemy7, i can't replicate the issue you are describing, it seems to be working fine on my end.<br>Can you make an attempt to replicate this on a blank project? it is sometimes hard to isolate an issue 

[Simulator Screen Recording - iPhone 15 Pro Max - 2024-06-25 at 16.39.43.mp4](https://uploads.linear.app/3241089d-736a-4422-900b-051e72accdbe/1b2a5dad-dcb1-47b3-95ad-082cf0b9048d/41c086b2-97a1-4c59-815f-0f32dba043da?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzMyNDEwODlkLTczNmEtNDQyMi05MDBiLTA1MWU3MmFjY2RiZS8xYjJhNWRhZC1kY2IxLTQ3YjMtOTVhZC0wODJjZjBiOTA0OGQvNDFjMDg2YjItOTdhMS00YzU5LTgxNWYtMGYzMmRiYTA0M2RhIiwiaWF0IjoxNzE5MzQ0Nzc5LCJleHAiOjMzMjg5OTA0Nzc5fQ.Qf5hO-1chrKkp3ost2AyleFgNbXUVaR-2qtjFSZ5mcU)

**Comment by bloemy7** [comment]
This reproduces from a blank project. It only reproduces if you get the data to come from a backend Firebase query, not with local data. I'll be sharing a video and a bug code very soon.

**Comment by bloemy7** [comment]
Here is the bug code for the blank project:
IT4ol/H60491xNhE17r+bPlVuyQvH182aLg7i8pBazwbfe+uB4kqevfOTERuTebiYWNieUyZq2oW0fuNhd71JPo5NQ+AfqpOwc9QWBbwbHmnRaatFJWgVHRCBfJgBXak0KSvkBFAWdZoSUY47GaUA/OvaBPtQeftDFU4Da/LZO4=

**Comment by bloemy7** [comment]
Uploading BugCovet.mp4…

**Comment by bloemy7** [comment]
https://github.com/FlutterFlow/flutterflow-issues/assets/1565896/9e6ea848-e72e-4602-9a7d-0cd720064ecf

**Comment by bloemy7** [comment]
https://github.com/FlutterFlow/flutterflow-issues/assets/1565896/03de3fef-d084-4d80-a529-de7d0955401b

**Comment by rzambroni** [comment]
i see, you are using conditional building and i was using conditional visibility that might be the reason i couldn't replicate it. Will take another look at your approach.

In the meantime if it is possible to use conditional visibility it appears to be working as expected and might help you workaround the issue for now.

<img src="https://uploads.linear.app/3241089d-736a-4422-900b-051e72accdbe/49204786-ec0d-42c6-8760-dba2f1fa450d/aef5eec2-5437-482c-ad52-a65b67973278?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzMyNDEwODlkLTczNmEtNDQyMi05MDBiLTA1MWU3MmFjY2RiZS80OTIwNDc4Ni1lYzBkLTQyYzYtODc2MC1kYmEyZjFmYTQ1MGQvYWVmNWVlYzItNTQzNy00ODJjLWFkNTItYTY1YjY3OTczMjc4IiwiaWF0IjoxNzE5NjA1NzczLCJleHAiOjMzMjkwMTY1NzczfQ.uyyoaMEWGa5Jw3pS-gY6kDdUlGZ8wTl-9_JVnHhDBwk " alt="image.png" width="599" height="568" />

**Comment by bloemy7** [comment]
The same thing happens with conditional visibility for us, it's not related to that it seems.
Are you using a Firebase query in the data table to reproduce it?
You can look at our sample project, it reproduces from scratch there. Any help would be very appreciated here since we are stuck for a big launch...

**Comment by rzambroni** [comment]
yep, I'm also querying a collection in Firebase, so it might be something else. Also tested web because i saw your video but was working fine, i'll continue checking.

**Comment by bloemy7** [comment]
![Captura de pantalla 2024-06-29 a la(s) 4 25 05 p m](https://github.com/FlutterFlow/flutterflow-issues/assets/1565896/aa4f8049-639b-494b-8828-e81c55e2d356)

I can reproduce this from total scratch, with the simplest of projects. It reproduces every single time. I don't understand what is happening? @rzambroni

**Comment by rzambroni** [comment]
can you add me to that from-scratch project as well? (rzambroni@somniosoftware.com)

**Comment by bloemy7** [comment]
Hi @rzambroni . I invited you to the project. Can we please prioritize this as it’s blocking a release for us. Thank you!

**Comment by Alezanello** [comment]
Hello!

I was able to reproduce the issue, although it is not consistent. Given its intermittent nature, I will forward this to the engineering team for a deeper investigation.

Best regards,<br>Azanello

**Comment by bloemy7** [comment]
Hi @Alezanello, is there any update or anything we can help with to make this advance? We're unfortunately blocked for an update due to this and would love any insights. Thanks!

**Comment by Alezanello** [comment]
Hi @bloemy7 !

The issue is currently in the engineering team's backlog. As soon as I have any updates, I will communicate them to you via this issue.

In the meantime, here's a potential workaround: In the second datatable query, try adding a trivial filter or an "Order By" clause that doesn't alter the results. Even if the filter or sorting doesn't change the returned data, it will be considered a different query, which may prevent the infinite loading issue. You can experiment with this approach to see if it resolves the problem for now.

**Comment by bloemy7** [comment]
@Alezanello we are doing only one query and using the results in both tables, so this shouldn't be the case. That is the weird part here. I'll try and move down to do query and change them to see if it works.

**Comment by bloemy7** [comment]
@Alezanello I've tried making it two separate queries and adding another orderBy but it's not fixing it. This is super frustrating... Please let me know.

**Comment by bloemy7** [comment]
Basically the only "workaround" that works is displaying both tables long enough so they properly initialize (multiple seconds), and then hiding one. But this is obviously not a viable production solution...

**Comment by bloemy7** [comment]
Hi, any fix or update here?

**Comment by bloemy7** [comment]
@Alezanello any update please?

**Comment by Franco98404** [comment]
Hello @Alezanello any news about this issue?

**Comment by Abi-rlv** [comment]
Hello @Alezanello, is there a fix for this issue ?

Actually, I got the same problem and I don't know how to solve this in another way cause I need to show 3 different DataTables conditionally by a state.

**Comment by bloemy7** [comment]
@Abi-rlv same issue here - I have not find any other workaround...

**Comment by Abi-rlv** [comment]
@bloemy7, I'm thinking about doing a page for each DataTable 😭

**Comment by sbis04** [comment]
The main reason this happens is because Flutter can't differentiate between the two DataTables since they accept the same type of object. The easiest way to resolve this issue is by specifying different `Value Key(s)` to the two or more similar `DataTable` widgets (you can find this inside the DataTable widget's right properties panel, under the `Testing` section)

<img width="303" alt="Image" src="https://github.com/user-attachments/assets/6e6251ea-b2f5-43a4-8183-fc45da5a780c" />

I have tested by adding the Value Keys and it seems to resolve the issue. Feel free to reopen if you still face the issue after adding the Value Keys.
